Fix: Blocks shifts by the border's size when removed or debug mode enabled

igrep opened the pull request:
# Problem

Blocks move right little by little when hit by the ball. It's not a problem when the number of blocks is small, but when the document contains many inline elements aligned in a block element, the rightmost element shifts significantly.

Here's an example reproduced by the `controlMode: "mouse"` feature created in https://github.com/canalun/brick-break-anywhere/pull/15, and hard-coding `visualizeBlocks: false` (checkout [my debug branch](https://github.com/igrep/brick-break-anywhere/tree/debug) if you reproduce by yourself):

![bba-unintentional-shift](https://github.com/canalun/brick-break-anywhere/assets/227057/c92bea31-033f-4c1d-a25d-fcebc19c1477)

As you can see, several right blocks shift so right that the ball breaks them without colliding. This is because the shifts make their actual positions differ from the positions managed by the game.

<details>
  <summary>The source of the screenshot HTML is here.</summary>

```html
<!DOCTYPE html>
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="Content-Language" content="ja">
  </head>
  <body>
    <span>0</span>
    <span>1</span>
    <span>2</span>
    <span>3</span>
    <span>4</span>
    <span>5</span>
    <span>6</span>
    <span>7</span>
    <span>8</span>
    <span>9</span>
    <span>a</span>
    <span>b</span>
    <span>c</span>
    <span>d</span>
    <span>e</span>
    <span>f</span>
    <span>g</span>
    <span>h</span>
    <span>i</span>
    <span>j</span>
    <span>k</span>
    <span>l</span>
    <span>m</span>
    <span>n</span>
    <span>o</span>
    <span>p</span>
    <span>q</span>
    <span>r</span>
    <span>s</span>
    <span>t</span>
    <span>u</span>
    <span>v</span>
    <span>w</span>
    <span>x</span>
    <span>y</span>
    <span>z</span>
    <span>A</span>
    <span>B</span>
    <span>C</span>
    <span>D</span>
    <span>E</span>
    <span>F</span>
    <span>G</span>
    <span>H</span>
    <span>I</span>
    <span>J</span>
    <span>K</span>
    <span>L</span>
    <span>M</span>
    <span>N</span>
    <span>O</span>
    <span>P</span>
    <span>Q</span>
    <span>R</span>
    <span>S</span>
    <span>T</span>
    <span>U</span>
    <span>V</span>
    <span>W</span>
    <span>X</span>
    <span>Y</span>
    <span>Z</span>
  </body>
</html>
```

</details>

# Solution

Use `outline` instead of `border` for highlighting block hit by the ball (or when debug mode).

# NOTE

I tested this change only in the example case and [my website's top page](https://the.igreque.info/). Tell me if you think of any side-effect.

igrep committed:
Fix: Blocks shifts by border when removed or debug mode enabled

canalun reviewed (commented):
Thank you so much...!!!! I've learnt `outline` from this PR.

I think the red-flush effect should be eye-catching, so what about using border with original width if the target has border and using outline if the target has no border?
In your implementation, if the target has border, the effect might be a bit unnoticeable.

igrep committed:
Highlight with border when the elemen has border

Thanks to @canalun's comment: "I think the red-flush effect should be eye-catching, so what about using border with original width if the target has border and using outline if the target has no border? In your implementation, if the target has border, the effect might be a bit unnoticeable."

https://github.com/canalun/brick-break-anywhere/pull/17#pullrequestreview-2193163709

igrep commented:
> I think the red-flush effect should be eye-catching, so what about using border with original width if the target has border and using outline if the target has no border?
In your implementation, if the target has border, the effect might be a bit unnoticeable.

Indeed! Do you mean this? https://github.com/canalun/brick-break-anywhere/pull/17/commits/0c8c4c1616edd90c4d19c695bfef80d3cb14ffae

igrep committed:
Small style fix

canalun commented:
Yes, exactly! I'm happy that you also think it's nice:)
Thank you for finding this bug and fixing it!!
